test_opensearchpy/test_server/test_clients.py

```diff
@@ -49,3 +49,15 @@ def test_bulk_works_with_bytestring_body(self) -> None:
 
         self.assertFalse(response["errors"])
         self.assertEqual(1, len(response["items"]))
+
+
+class TestClose(OpenSearchTestCase):
+    def test_close_doesnt_break_client(self) -> None:
+        self.client.cluster.health()
+        self.client.close()
+        self.client.cluster.health()
+
+        with self.client as client:
```

Comment:
Split into its own test, tests are cheap.

Maybe a range iteration 2 times is nicer? I wish we had the Ruby equivalent of `2.times do`. Now I'm really nitpicking :)